### ISSUE TITLE
Decrypting with N prime

### ISSUE THREAD

**Issue by Signum21** [RsaCtfTool/RsaCtfTool#313]
This is probably extremely rare but if someone uses a prime N as modulus we can easily obtain the private key to decipher the message
The Phi of a prime number is N-1

Python example code:
```Python
from pwn import *
from Crypto.Util.number import *

#n is prime
n = 171731371218065444125482536302245915415603318380280392385291836472299752747934607246477508507827284075763910264995326010251268493630501989810855418416643352631102434317900028697993224868629935657273062472544675693365930943308086634291936846505861203914449338007760990051788980485462592823446469606824421932591                                                                  
e = 65537
plaintext = b"this_is_a_test"
encrypted = pow(bytes_to_long(plaintext), e, n)

phi = n-1
d = pow(e, -1, phi)
decrypted = pow(encrypted, d, n)
decryptedText = unhex(hex(decrypted)[2:]).decode()
print(decryptedText)
```

**Comment by daedalus** [comment]
If N is prime, there is no use case for RSA, that is a very trivial case.

**Comment by Signum21** [comment]
Well, this is a tool for CTF, in real life obviosly that's not gonna happen but in a CTF everything is possible.

**Comment by wwallace** [comment]
I would say this would be useful as N prime but also to check if N is a Prime squared, Cubed, etc. Currently factordb attack implementation doesn't seem quite work with this scenario:
```
N = 1128137999850045612492145429133282716267233566834715456536184965477269592934207986950131365518741418540788596074115883774105736493742449131477464976858161587355643311888741515506653603321337485523828144179637379528510277430032789458804637543905426347328041281785616616421292879871785633181756858096548411753919440011378411476275900648915887370219369154688926914542233244450724820670256654513052812215949495598592852131398736567134556141744727764716053145639513031
E = 65537
```
N is a cube of prime http://factordb.com/index.php?id=1100000001449359491

borrowing @Signum21 's example but with N = Prime^3:

```
plaintext = b"this_is_a_test"
encrypted = pow(bytes_to_long(plaintext), pk.e, pk.n)

p = 10410080216253956216713537817182443360779235033823514652866757961082890116671874771565125457104853470727423173827404139905383330210096904014560996952285911
phi = p * p * (p-1)
d = pow(pk.e, -1, phi)

decrypted = pow(encrypted, d, pk.n)
decryptedText = unhex(hex(decrypted)[2:]).decode()
print(decryptedText)
```

**Comment by daedalus** [comment]
Hi you don't need to perform a factordb lookup when you have a integer in the form p^n where p is prime and n > 0.
You just need to calculate the nth root of that integer with the base set to the prime.

**Comment by Ganapati** [comment]
Hi, thanks for this finding.
I have no time to patch it, can you make a pull request ?

**Comment by daedalus** [comment]
@Ganapati is it ok to close it?

**Comment by daedalus** [comment]
Fixed in the PR #321
